### DISCUSSION TITLE
Anywhere to discuss getting virtual ins / outs to work on Windows?

### DISCUSSION THREAD

**Discussion by aidan959** [Boddlnagg/midir#159] [General]
I'm looking to use virtual ins and outs on Windows for a project and see its not supported. Is there any thread or branch to look at where work is being done on this or is it not in scope? Am happy to contribute on this issue if possible.

**Comment by Boddlnagg** [comment]
Virtual ports on Windows are unfortunately not supported by the Windows MIDI infrastructure (unless you use special third-party drivers). See also https://github.com/Boddlnagg/midir/issues/37 and https://github.com/Boddlnagg/midir/issues/149.

Microsoft is currently developing a new MIDI infrastructure which will bring support for not only MIDI 2.0 but also virtual ports. However, it's not ready yet, see https://github.com/microsoft/midi.
